src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs

```diff
@@ -468,36 +508,13 @@ public Task DeleteChannel(int channelId, bool force)
-            if (channel.Description != null) { updateParameters.Add(new Parameter("channel_description", channel.Description)); }
-            if (channel.Password != null) { updateParameters.Add(new Parameter("channel_password", channel.Password)); }
-            if (channel.Codec != null) { updateParameters.Add(new Parameter("channel_codec", (int)channel.Codec)); }
-            if (channel.CodecQuality != null) { updateParameters.Add(new Parameter("channel_codec_quality", channel.CodecQuality)); }
-            if (channel.MaxClients != null) { updateParameters.Add(new Parameter("channel_maxclients", channel.MaxClients)); }
-            if (channel.MaxFamilyClients != null) { updateParameters.Add(new Parameter("channel_maxfamilyclients", channel.MaxFamilyClients)); }
-            if (channel.Order != null) { updateParameters.Add(new Parameter("channel_order", channel.Order)); }
-            if (channel.IsPermanent != null) { updateParameters.Add(new Parameter("channel_flag_permanent", channel.IsPermanent)); }
-            if (channel.IsSemiPermanent != null) { updateParameters.Add(new Parameter("channel_flag_semi_permanent", channel.IsSemiPermanent)); }
-            if (channel.IsTemporary != null) { updateParameters.Add(new Parameter("channel_flag_temporary", channel.IsTemporary)); }
-            if (channel.IsDefaultChannel != null) { updateParameters.Add(new Parameter("channel_flag_default", channel.IsDefaultChannel)); }
-            if (channel.IsMaxClientsUnlimited != null) { updateParameters.Add(new Parameter("channel_flag_maxclients_unlimited", channel.IsMaxClientsUnlimited)); }
-            if (channel.IsMaxFamilyClientsUnlimited != null) { updateParameters.Add(new Parameter("channel_flag_maxfamilyclients_unlimited", channel.IsMaxFamilyClientsUnlimited)); }
-            if (channel.IsMaxFamilyClientsInherited != null) { updateParameters.Add(new Parameter("channel_flag_maxfamilyclients_inherited", channel.IsMaxFamilyClientsInherited)); }
-            if (channel.NeededTalkPower != null) { updateParameters.Add(new Parameter("channel_needed_talk_power", channel.NeededTalkPower)); }
-            if (channel.PhoneticName != null) { updateParameters.Add(new Parameter("channel_name_phonetic", channel.PhoneticName)); }
-            if (channel.IconId != null) { updateParameters.Add(new Parameter("channel_icon_id", (int)channel.IconId)); }
-            if (channel.IsCodecUnencrypted != null) { updateParameters.Add(new Parameter("channel_codec_is_unencrypted", channel.IsCodecUnencrypted)); }
-            if (channel.ParentChannelId != null) { updateParameters.Add(new Parameter("channel_cpid", channel.ParentChannelId)); }
-
-            return Client.Send("channeledit", updateParameters.ToArray());
+            var res = await Client.Send("channeledit",
+                new Parameter("cid", channel_id),
+                new Parameter(EditChannel.ToString(), value))
+                .ConfigureAwait(false);
```

Comment:
Is this safe to remove?

Reply:
I think so, I tested all frags from ChannelEdit and it works fine.

Reply:
How do you set the Name and the Topic using one command?

Reply:
I'm currently working on this and also on other improvements, for now it's limited to one parameter per command.

Reply:
What is the reason to do it like this?

Reply:
Error handling, if something goes wrong with one command and parameter I'm sure what and when.

Reply:
I think we can find a better solution that doesn't break API compatibility. I'll see if I can come up with something.